std/utf.d

```diff
@@ -187,7 +187,7 @@ body
 {
     import core.bitop : bsr;
     immutable msbs = 7 - bsr(~c);
-    if (msbs < 2 || msbs > 4)
+    if (!~c || msbs < 2 || msbs > 4)
```

Comment:
correct me if I'm wrong: isn't !~c equal to c == 0. you comment sort of also implies that.